packages/backend/src/managers/playground/McpServerManager.spec.ts

```diff
@@ -0,0 +1,124 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
+import fs from 'node:fs';
+import os from 'node:os';
+import path from 'node:path';
+import type { McpServer } from './McpServerManager';
+import { McpServerType, McpServerManager } from './McpServerManager';
+
+/* eslint-disable sonarjs/no-nested-functions */
+describe('McpServerManager', () => {
+  let appUserDirectory: string;
+  let consoleErrors: string;
+  let consoleWarnings: string;
+  beforeEach(async () => {
+    appUserDirectory = await fs.promises.mkdtemp(path.join(os.tmpdir(), 'mcp-server-manager-test-'));
+    consoleErrors = '';
+    vi.spyOn(console, 'error').mockImplementation((...args) => {
```

Comment:
I would not really recommend to test console.* functions, logging information could change be updated, and IHMO should not be use to test the flow of the tested code

Reply:
If you need to check for an error, if should be raised, if this is only warning, we don't need to test it, this is not core feature

Reply:
Well, here I'm testing it as part of the behavior.
If it's not a core feature, then I'd suggest not to add it.
Not sure if you've noticed, but I'm kind of orthodox and extremely strict with tests.
If there's something in the production code I will test it. If that's not part of the impl, then I'll remove it from the production code. But, in general, I won't have both.

Reply:
Yeah and I really appreciate the quality of code you provide, but testing console.log makes the test too complex for just checking that something will appear in the terminal

Reply:
I will remove this, but note that I'm extremely uncomfortable doing so.